server/src/utils/socketIoUtils.ts

```diff
@@ -116,6 +119,5 @@ export const emitRoadmapEvent = async <T extends ClientEvents>(
 
     // Sockets are added to a room with their associated user id when they connect to allow sending like this
     io.to(`${user.id}`).emit(event, ...parameters);
```

Comment:
```suggestion
    io.to(usersWithPermission.map((user) =>`${user.id}`).emit(event, ...parameters);
```

the `.to()` method can also take an array, I'm not sure if it would have benefits, but it would be an option.